python/pyspark/ml/param/shared.py

```diff
@@ -814,3 +814,25 @@ def getDistanceMeasure(self):
         """
         return self.getOrDefault(self.distanceMeasure)
 
+
+class HasValidationIndicatorCol(Params):
```

Comment:
Would you mind running the codegen again, like this command for example
`pushd python/pyspark/ml/param/ && python _shared_params_code_gen.py > shared.py && popd` and push the result if there is a diff?  I think the DecisionTreeParams should be at the bottom of the file..

Reply:
You are right. DecisionTreeParams should be at the bottom.